### PR TITLE
Implementation of new MCDA types

### PR DESCRIPTION
**Pull request by ConnectedSystems**
As in title.

Addresses the first point in the list tracked by #677

### CODE REVIEW

File: src/decision/Criteria/SRMCriteria.jl [hunk 1]
```diff
@@ -0,0 +1,69 @@
+"""
```
Comment (Rosejoycrocker) [inline src/decision/Criteria/SRMCriteria.jl:1]: Will this be removed if shading is an environmental layer ? Or is this different?

Comment (ConnectedSystems) [reply src/decision/Criteria/SRMCriteria.jl:1]: Yes, in fact it's removed in the next set of changes

Comment (Rosejoycrocker) [reply src/decision/Criteria/SRMCriteria.jl:1]: I'm guessing tests will be updated in other changes also?

**Review by Rosejoycrocker — APPROVED** [review]
Looks good to me

File: src/decision/Criteria/SRMCriteria.jl (hunk 1 above)
Comment (ConnectedSystems) [reply src/decision/Criteria/SRMCriteria.jl:1]: Yes ma'am

File: src/ExtInterface/ADRIA/Domain.jl [hunk 2]
```diff
@@ -1,3 +1,13 @@
+using ADRIA.decision:
+    DecisionThresholds,
+    DecisionWeights,
+    DepthThresholds
+
+using ADRIA.decision:
+    SeedCriteriaWeights,
+    FogCriteriaWeights,
+    SRMCriteriaWeights
```
[the remark this answers is not in the document]
Comment (ConnectedSystems) [reply src/ExtInterface/ADRIA/Domain.jl:9]: I'm separating them conceptually. The first is Decision-based types, the second are the weights.

Note the slight deviation from Blue style which says the import alphabetically. The first follows this, but the second follows the usual order used throughout ADRIA (seed, fog, shade)

File: src/decision/Criteria/DecisionPreferences.jl [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    decision_matrix(loc_names::Vector{T}, criteria_names::Vector{T}, criteria_vals::Matrix)::YAXArray where {T<:Union{String,Symbol}}
+
+Construct a decision matrix.
+
+# Arguments
+- `loc_names` : location names
+- `criteria_names` : name of criteria being considered
+- `criteria_vals` : values for each criteria
+"""
+function decision_matrix(
```
[the remark this answers is not in the document]
Comment (ConnectedSystems) [reply src/decision/Criteria/DecisionPreferences.jl:24]: The decision matrix is the data meaning. That's what it is in the context of MCDA.

File: src/decision/Criteria/DecisionPreferences.jl [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    catch err
+        if err isa DomainError
+            # Return empty matrix to signify no ranks
+            return [;;]
+        end
+
+        rethrow(err)
+    end
+
+    return [collect(getAxis(:location, dm)[rank_idx][1:n_locs]) rank_idx[1:n_locs]]
```
[the remark this answers is not in the document]
Comment (ConnectedSystems) [reply src/decision/Criteria/DecisionPreferences.jl:141]: Thanks!

File: src/decision/Criteria/DecisionPreferences.jl [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+# Arguments
+- `criteria_name` : criteria to apply thresholds to
+- `threshold` : lower and upper bounds of values to retain (inclusive)
+- `dm` : a decision matrix
+
+# Returns
+A new YAXArray
+"""
+function apply_threshold(
```
[the remark this answers is not in the document]
Comment (ConnectedSystems) [reply src/decision/Criteria/DecisionPreferences.jl:158]: I think this set of functions were unnecessary in the end. Likely I will remove in the next set of PRs anyway.

File: src/decision/Criteria/DecisionPreferences.jl [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+        @warn msg
+        return dm
+    end
+
+    target_criteria = dm.criteria .== criteria_name
+    target_vals = dm.data[:, target_criteria]
+
+    valid_locs = vec(threshold[1] .<= target_vals .<= (threshold[1]+threshold[2]))
+
+    return dm[valid_locs, :]
```
[the remark this answers is not in the document]
Comment (ConnectedSystems) [reply src/decision/Criteria/DecisionPreferences.jl:173]: As elsewhere, I thought it doesn't work with bitvector/booleans

File: src/decision/Criteria/DecisionPreferences.jl [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+
+"""
+    apply_depth_threshold(dom, params, decision_mat)
+
+Apply a depth thresholds using values from a parameter set.
+
+# Arguments
+- `dom` : Domain
+- `params` : Parameter specification
+- `decision_mat` : Decision matrix to update
```
[the remark this answers is not in the document]
Comment (ConnectedSystems) [reply src/decision/Criteria/DecisionPreferences.jl:184]: True, but as said elsewhere, in the end I don't use this so have removed in the next set of PRs.

File: src/decision/Criteria/DecisionPreferences.jl [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+    loc_names::Vector{T},
+    criteria_names::Vector{T2},
+    criteria_vals::Matrix
+)::YAXArray where {T<:Union{String,Symbol},T2<:Union{String,Symbol}}
+    axlist = (
+        Dim{:location}(loc_names),
+        Dim{:criteria}(criteria_names),
+    )
+
+    return YAXArray(
```
[the remark this answers is not in the document]
Comment (ConnectedSystems) [reply src/decision/Criteria/DecisionPreferences.jl:34]: I'll leave this as is for now, maybe update in next PR.

File: src/decision/Criteria/DecisionPreferences.jl (hunk 6 above)
[the remark this answers is not in the document]
Comment (Zapiano) [reply src/decision/Criteria/DecisionPreferences.jl:173]: Just to be clear, I'm not saying you need to change this.

[the remark this answers is not in the document]
Comment (ConnectedSystems) [reply src/decision/Criteria/DecisionPreferences.jl:173]: I like it, it is clearer

**Comment by ConnectedSystems** [comment]
@Zapiano ready for second review.

After this set of changes in, I may rebase onto the `yax-main` branch.

**Comment by ConnectedSystems** [comment]
I'm just going to go ahead and merge - thanks for the comments.

**Pull request merged**